main/main.c

```diff
@@ -1932,6 +1932,11 @@ void php_request_shutdown(void *dummy)
 		zend_unset_timeout();
 	} zend_end_try();
 
+	/* 17. Deactivate Zend signals */
+#ifdef ZEND_SIGNALS
+	zend_signal_deactivate();
+#endif
+
```

Comment:
I think, it's not a good idea to add several syscalls on each request. May be enable this only in DEBUG build?

Reply:
Maybe just drop the SIGNAL_BEGIN_CRITICAL in zend_signal_deactivate? Then there shouldn't be any syscalls, and I don't see a reason why this needs to block signals during the reset, as long as active=0 is set first.

Reply:
agree